charts/internal/shoot-system-components/charts/csi-driver-node/templates/daemonset.yaml

```diff
@@ -48,6 +48,9 @@ spec:
         {{- if .Values.driver.volumeAttachLimit }}
         - --volume-attach-limit={{ .Values.driver.volumeAttachLimit }}
         {{- end }}
+        {{- if .Values.driver.legacyXFS }}
+        - --legacyXFS={{ .Values.driver.legacyXFS }}
```

Comment:
Also this flag is not even correct. https://github.com/kubernetes-sigs/aws-ebs-csi-driver/blob/7fd319a43ee6fe339517051226c506304d3f5f94/pkg/driver/options.go#L120 is the proper format.